core/trino-main/src/main/java/io/trino/server/ServerMainModule.java

```diff
@@ -310,12 +302,6 @@ protected void setup(Binder binder)
         binder.bind(PagesIndex.Factory.class).to(PagesIndex.DefaultFactory.class);
         newOptionalBinder(binder, OperatorFactories.class).setDefault().to(TrinoOperatorFactories.class).in(Scopes.SINGLETON);
 
-        jsonCodecBinder(binder).bindJsonCodec(TaskStatus.class);
```

Comment:
these are not needed for serialization on worker?

Reply:
`JsonCodecBinder` is used if you need to inject a `JsonCodec`, such as for use with `HttpClient`. If you're only returning an object from a JAX-RS method, then you don't need it, since our `JsonMapper` will do it directly using `ObjectMapper`.